posthog/session_recordings/session_recording_api.py

```diff
@@ -394,13 +398,29 @@ def safely_get_object(self, queryset) -> SessionRecording:
         return recording
 
     def list(self, request: request.Request, *args: Any, **kwargs: Any) -> Response:
-        query = filter_from_params_to_query(request.GET.dict())
+        user_distinct_id = cast(User, request.user).distinct_id
 
-        self._maybe_report_recording_list_filters_changed(request, team=self.team)
-        return list_recordings_response(
-            list_recordings_from_query(query, cast(User, request.user), team=self.team),
-            context=self.get_serializer_context(),
-        )
+        try:
+            query = filter_from_params_to_query(request.GET.dict())
+
+            self._maybe_report_recording_list_filters_changed(request, team=self.team)
+            return list_recordings_response(
+                list_recordings_from_query(query, cast(User, request.user), team=self.team),
+                context=self.get_serializer_context(),
+            )
+
+        except ValidationError:
+            raise
+        except CHQueryErrorTooManySimultaneousQueries:
+            raise Throttled(detail="Too many simultaneous queries. Try again later.")
+        except (ServerException, Exception) as e:
+            if isinstance(e, ServerException) and "CHQueryErrorTimeoutExceeded" in str(e):
+                raise Throttled(detail="Query timeout exceeded. Try again later.")
+
+            posthoganalytics.capture_exception(
```

Comment:
Why aren't CH query errors and validation errors captured as exceptions? For too many queries (429?) it's understandable-ish (probably is visible in some specified dashboard), but not sure how the validation error can appear and why not to capture it 👀

Reply:
I am 99% sure that validation error is handled correctly elsewhere in the stack

and then also don't want to hammer our event ingestion for errors that are somewhat expected

it's what reminded me of this throttling idea https://posthog.slack.com/archives/C07AA937K9A/p1744368735181649